test/Crystallization.cpp

```diff
@@ -0,0 +1,177 @@
+{
+	const std::string& modelFilePath = std::string("/data/configuration_PBM_DPFR_primarySecondaryNucleationGrowth_benchmark1.json");
+	const std::string& refFilePath = std::string("/data/ref_PBM_DPFR_primarySecondaryNucleationGrowth_benchmark1.h5");
+	const std::vector<double> absTol = { 7e+6 };
+	const std::vector<double> relTol = { 1e-10 };
+
+	cadet::test::column::FVparams disc(25);
+	cadet::test::column::testReferenceBenchmark(modelFilePath, refFilePath, "001", absTol, relTol, disc, true);
+}
+
+// Note: two entries fail, given in (row,col): (204, 190), (204, 191)
+TEST_CASE("Crystallization Jacobian verification for a CSTR with initial distribution and growth", "[CrysToFix1]") // "[Crystallization],[UnitOp],[Jacobian]")
+{
+	// read json model setup file
+	const std::string& modelFileRelPath = std::string("/data/configuration_PBM_CSTR_growth_benchmark1.json");
+	const std::string setupFile = std::string(getTestDirectory()) + modelFileRelPath;
+	cadet::JsonParameterProvider pp_setup(cadet::JsonParameterProvider::fromFile(setupFile));
+
+	pp_setup.pushScope("model");
+	pp_setup.pushScope("unit_001");
+
+	cadet::test::column::testJacobianAD(pp_setup, std::numeric_limits<float>::epsilon() * 100.0);
+}
+
+// Note: two entries fail, given in (row,col): (204, 190), (204, 191)
+TEST_CASE("Crystallization Jacobian verification for a CSTR with initial distribution and size-dependent growth", "[CrysToFix2]") // "[Crystallization],[UnitOp],[Jacobian]")
+{
+	// read json model setup file
+	const std::string& modelFileRelPath = std::string("/data/configuration_PBM_CSTR_growthSizeDep_benchmark1.json");
+	const std::string setupFile = std::string(getTestDirectory()) + modelFileRelPath;
+	cadet::JsonParameterProvider pp_setup(cadet::JsonParameterProvider::fromFile(setupFile));
+
+	pp_setup.pushScope("model");
+	pp_setup.pushScope("unit_001");
+
+	cadet::test::column::testJacobianAD(pp_setup, std::numeric_limits<float>::epsilon() * 100.0);
+}
+
+// Note: two entries fail, given in (row,col): (204, 190), (204, 191)
+TEST_CASE("Crystallization Jacobian verification for a CSTR with primary nucleation and growth", "[CrysToFix3]") // "[Crystallization],[UnitOp],[Jacobian]")
+{
+	// read json model setup file
+	const std::string& modelFileRelPath = std::string("/data/configuration_PBM_CSTR_primaryNucleationAndGrowth_benchmark1.json");
+	const std::string setupFile = std::string(getTestDirectory()) + modelFileRelPath;
+	cadet::JsonParameterProvider pp_setup(cadet::JsonParameterProvider::fromFile(setupFile));
+
+	pp_setup.pushScope("model");
+	pp_setup.pushScope("unit_001");
+
+	cadet::test::column::testJacobianAD(pp_setup, std::numeric_limits<float>::epsilon() * 100.0);
+}
+
+// Some more comparisons fail
+TEST_CASE("Crystallization Jacobian verification for a CSTR with primary nucleation, growth and growth rate dispersion", "[CrysToFix4]") // "[Crystallization],[UnitOp],[Jacobian]")
```

Comment:
@WFlynnZ For this test case, i.e. when we include growth rate dispersion, we can some more errors in the Jacobian